src/main/resources/com/openshift/jenkins/plugins/OpenShiftDSL.groovy

```diff
@@ -267,7 +270,7 @@ class OpenShiftDSL implements Serializable {
 
     @NonCPS
     private void dieIfNotWithin(ContextId me, Context context, ContextId required) throws AbortException {
-        if (contextContains(context, required)) {
+        if (!contextContains(context, required)) {
```

Comment:
yeah ... the fix seems right  ... a bit surprised this did not come up sooner ... prior to your change here that line had not been touched since Oct 25 2017

good catch!!

I'm half wondering if people have been getting away with a withProject only, where somehow it defaulted to project jenkins is running in, and will get bitten by your change.

Some simple test cases added to openshift/origin or even at least the examples dir here for reference wouldn't be a bad idea, just to get the warm fuzzy ... also would be good to compare the results of that simple test case before and after ... assuming it would reproduce what issue you and @waveywaves were hitting @akram